tests/test_text.py

```diff
@@ -0,0 +1,775 @@
+
+        # Check whether the number of columns in `selected_features' is equal to num_features selected
+        # by solving set cover
+        self.assertEqual(selected_features.shape[1], 2)
+
+        # Verify the selected indices
+        self.assertListEqual(list(selected_features.columns), ['item2', 'item3'])
+        self.assertTrue(set(selected_features.columns).issubset(data.columns))
+
+    ################################################
+    ########## Tests for greedy selection ##########
+    ################################################
+
+    # Check selection for a single labels column of ones (select column with one)
+    # Check selection for infeasible instance (empty)
+    # Check selection for label matrix with considerable coverage (same features should select with the same seed)
+    def test_text_based_greedy_num_feature_one_or_infeasible_or_max(self):
+        # Fix the random seed
+        seed = 12
+        np.random.seed(seed)
+
+        data = pd.DataFrame(
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+
+        ### Labels with a single column with all ones ###
+        labels_single_col = pd.DataFrame({"item1": [1, 0, 0, 0, 0], "item2": [1, 0, 0, 0, 0], "item3": [1, 0, 0, 0, 0],
+                                          "item4": [1, 0, 0, 0, 0], "item5": [1, 0, 0, 0, 0]})
+
+        method_single_col = SelectionMethod.TextBased(num_features=1,
+                                                      optimization_method="greedy",
+                                                      cost_metric="unicost")
+
+        # Verify that the number of columns is data and labels match
+        self.assertEqual(data.shape[1], labels_single_col.shape[1])
+
+        selector = Selective(method_single_col, seed=seed)
+        selector.fit(data, labels_single_col)
+        selected_features_single_col = selector.transform(data)
+
+        assert selector.selection_method.trials == 10
+        self.assertTrue(isinstance(selected_features_single_col, pd.DataFrame))
+
+        # Verify greedy selects the item1
+        self.assertListEqual(list(selected_features_single_col.columns), ['item1'])
+
+        ### Labels for infeasible instance ###
+        labels_infeasible = pd.DataFrame({"item1": [0, 0, 0, 0, 0], "item2": [0, 0, 0, 0, 0], "item3": [0, 0, 0, 0, 0],
+                                          "item4": [0, 0, 0, 0, 0], "item5": [0, 0, 0, 0, 0]})
+
+        method_infeasible = SelectionMethod.TextBased(num_features=2,
```

Comment:
learnign questions; when the labels is infeasible, what do we actually return? Is the final selection is just empty? or sth like that?

Reply:
When the labels are infeasible, the selector will return an empty dataframe as the final selection because no feature can be selected that covers all labels.

Reply:
Is this captured in the pydoc? We should add a comment about infeasible behavior